fem/fespace.cpp

```diff
@@ -615,6 +670,62 @@ FiniteElementSpace::AddDependencies(SparseMatrix& deps, Array<int>& master_dofs,
    }
 }
 
+void FiniteElementSpace
+::AddEdgeFaceDependencies(SparseMatrix &deps, Array<int> &master_dofs,
+                          const FiniteElement *master_fe,
+                          Array<int> &slave_dofs, int slave_face,
+                          const DenseMatrix *pm) const
+{
+   // In variable-order spaces in 3D, we need to only constrain interior face
+   // DOFs (this is done one level up), since edge dependencies can be more
+   // complex and are primarily handled by edge-edge dependencies. The one
+   // exception is edges of slave faces that lie in the interior of the master
+   // face, which are not covered by edge-edge relations. This function finds
+   // such edges and makes them constrained by the master face.
+   // See also https://github.com/mfem/mfem/pull/1423#issuecomment-633916643
+
+   Array<int> V, E, Eo; // TODO: LocalArray
```

Comment:
What is `LocalArray` and why `TODO`?

Reply:
I envision a descendant of `Array` for internal use like here, that would use a small buffer local to the class to hold its elements. Using `LocalArray` would avoid heap allocations. It's a common use case that we need small temporary arrays in methods like this, and a large number of heap allocations may hurt performance.